appmgr/src/net/mdns.rs

```diff
@@ -107,38 +108,43 @@ impl MdnsControllerInner {
                 std::ffi::CStr::from_ptr(self.hostname_raw)
             );
             for key in self.services.values() {
-                let lan_address = key
-                    .public()
-                    .get_onion_address()
-                    .get_address_without_dot_onion()
-                    + ".local";
-                tracing::debug!("Adding mdns CNAME entry for {}", &lan_address);
-                let lan_address_ptr = std::ffi::CString::new(lan_address)
-                    .expect("Could not cast lan address to c string");
-                res = avahi_sys::avahi_entry_group_add_record(
-                    self.entry_group,
-                    avahi_sys::AVAHI_IF_UNSPEC,
-                    avahi_sys::AVAHI_PROTO_UNSPEC,
-                    avahi_sys::AvahiPublishFlags_AVAHI_PUBLISH_USE_MULTICAST
-                        | avahi_sys::AvahiPublishFlags_AVAHI_PUBLISH_ALLOW_MULTIPLE,
-                    lan_address_ptr.as_ptr(),
-                    avahi_sys::AVAHI_DNS_CLASS_IN as u16,
-                    avahi_sys::AVAHI_DNS_TYPE_CNAME as u16,
-                    avahi_sys::AVAHI_DEFAULT_TTL,
-                    self.hostname.as_ptr().cast(),
-                    self.hostname.len(),
+                self.load_one(key);
+            }
+        }
+    }
+    fn load_one(&self, key: &TorSecretKeyV3) {
+        unsafe {
```

Comment:
Doesn't unsafe require the function be annotated too (Safety)?
https://doc.rust-lang.org/std/keyword.unsafe.html

Reply:
that would make every function in rust unsafe